### PR TITLE
feat(ui): add info to "No results found"

### PR DESCRIPTION
**Pull request by alxu-sentry**
Currently, the Performance Summary defaults to a 24-hour filter. The "No
results found" message made me think that the data had disappeared when
in fact it was simply filtered out.

Renaming this to "No results found for the given query" is an easy way
to give users a hint that they should investigate the filters.

I only tested this for the performance page, not the other uses of
GridEditable, but I think it applies for all of them.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx [hunk 1]
```diff
@@ -368,7 +368,7 @@ class GridEditable<
       <GridRow>
         <GridBodyCellStatus>
           <EmptyStateWarning>
-            <p>{t('No results found')}</p>
+            <p>{t('No results found for the given query')}</p>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: Maybe

> No results found for your query

Language feels a bit nicer

Comment (alxu-sentry) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: I feel like "your" implies that the user is responsible for the query, but here the query may not be explicitly provided by the user, like in my initial example of "I just navigated to Performance, what do you mean 'my query'?".

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: Yeah that's true. Something about this wording just feels a little odd to me, but I can't quite place my finger

Comment (silent1mezzo) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: Should a user ever get to a page where there are no results? @EvanPurkhiser works nicely is we assume that the only way they'll see it is if they change the filters/query.

Comment (alxu-sentry) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: Not sure what you're asking here. Performance defaulting to last 24 hours was an example of Sentry automatically inserting a filter for me.

Comment (alxu-sentry) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: @silent1mezzo ^

**Comment by alxu-sentry** [comment]
I wonder if here we could display a CTA like `Try <a>selecting a different project</a> or <a>adjusting the time range</a>.` with the former only showing up when not all projects are selected and the latter only showing up when the time range is not 90 days.

**Comment by dashed** [comment]
@alxu-sentry Maybe "We could not find any events in the [last 24 hours]. Try a different time range."

**Comment by alxu-sentry** [comment]
> @alxu-sentry Maybe "We could not find any events in the [last 24 hours]. Try a different time range."

"No results found for [the given query] in [the X project/the selected projects] [over the last 24 hours]"? sounds like a huge pain to translate though...

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx (hunk 1 above)
Comment (silent1mezzo) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: @alxu-sentry Put a different way, landing on performance should never result in `No results found`. That should be true everywhere in the app unless you change it, in which case it's your query that caused no results to be found.

**Comment by silent1mezzo** [comment]
> "No results found for [the given query] in [the X project/the selected projects] [over the last 24 hours]"? sound

It's too verbose as well, regardless of translations. 95% of the time when someone doesn't find something it's due to an incorrect time filter. @dashed suggestion works but it also needs to handle the cases where someone has selected an absolute time.

**Comment by alxu-sentry** [comment]
> > "No results found for [the given query] in [the X project/the selected projects] [over the last 24 hours]"? sound
> 
> It's too verbose as well, regardless of translations. 95% of the time when someone doesn't find something it's due to an incorrect time filter. @dashed suggestion works but it also needs to handle the cases where someone has selected an absolute time.

The original idea behind this PR is: in some cases the user may not have explicitly selected a filter. For example, when first visiting the performance page, by default the data is limited to the past 24 hours, even if there is no data in that period but there is data before that. The user is then confused because they haven't intentionally specified any filter, so they assume that "No results found" means that all their data is gone.

In that light, I think this suggestion works as long as the only filter that is automatically specified is time.

However, I noticed that the project can be automatically specified in at least one circumstance: if you have multiple projects selected (e.g. My Projects default under multi-project view), then select an event, then return to the performance landing page without using the back button, then the displayed data will be limited to the project of the previously selected event. Similarly, the implicit filter previously specified by the Display mode will be added to the explicitly specified filters.

I think if we can fix both of these, then displaying only the time CTA would be appropriate.

**Comment by silent1mezzo** [comment]
@alxu-sentry We're spending too much time nitpicking on this change.

Either go with `No results found for your query` or `No results found [in the last 24 hours|between x and y]`.

**Comment by alxu-sentry** [comment]
> @alxu-sentry We're spending too much time nitpicking on this change.
> 
> Either go with `No results found for your query` or `No results found [in the last 24 hours|between x and y]`.

I'm strongly in favor of "No results found for your query" until we can fix global project selection bugs, and revisit the message then. Just need someone to mash that Approve button :)

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx (hunk 1 above)
Comment (silent1mezzo) [inline src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]:
```suggestion
            <p>{t('No results found for your query')}</p>
```

**Review by silent1mezzo — APPROVED** [review]
Added the agreed-upon wording in another message. Once that's add added we're good to go.

File: src/sentry/static/sentry/app/components/gridEditable/index.tsx (hunk 1 above)
Comment (alxu-sentry) [reply src/sentry/static/sentry/app/components/gridEditable/index.tsx:371]: bleh i think i actually already expressed opposition to this but i don't want to argue about it anymore so it's fine for now

**Pull request merged**